File: python/260.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        x = 0
        for n in nums:
            x ^= n
        x = x & (~x+1)
        y1 = y2 = 0
        for n in nums:
            if n & x:
                y1 ^= n
            else:
                y2 ^= n
        return y1, y2

File: cpp/260.cpp
#include<vector>
using namespace std;

class Solution {
public:
  vector<int> singleNumber(vector<int>& nums) {
    int x = 0;
    for(int n : nums)
      x ^= n;
    x = x & (~x + 1);
    int y1=0, y2=0;
    for(int n : nums)
      if(x ^ n)
        y1 ^= n;
      else
        y2 ^= n;
    return {y1, y2};
  }
};
